oauthproxy.go

```diff
@@ -1172,6 +1182,53 @@ func extractAllowedGroups(req *http.Request) map[string]struct{} {
 	return groups
 }
 
+func (p *OAuthProxy) checkOpenPolicyAgent(req *http.Request, session *sessionsapi.SessionState) bool {
+	if p.openPolicyAgentURL == "" {
+		return true
+	}
+
+	query := map[string]string{}
+	for k := range req.URL.Query() {
+		query[k] = req.URL.Query().Get(k)
+	}
+
+	var request struct {
+		Input struct {
+			Query map[string]string `json:"query,omitempty"`
+			Token string            `json:"token"`
```

Comment:
Out of interest, why just the token and not the whole session object?

Reply:
I use the token because it contains all required information (user, e-mail, groups, ..). Alternative we could pass the session information including or excluding the token to avoid the decode of the token in OPA.